File: app/src/main/java/com/rr/tvstarterkit/data/preference/BooleanUtils.java
package com.rr.tvstarterkit.data.preference;

import android.text.TextUtils;
import android.util.Log;


public final class BooleanUtils {

    private static final String TAG = "BooleanUtils";

    /**
     * @param input
     * @return true if input is one of (T, True, TRUE, Y, yes, YES, 1, OK, ok, on, ON)
     */
    public static boolean isTrue(String input) {
        boolean result = false;
        try {
            if (!TextUtils.isEmpty(input)) {
                input = input.toLowerCase();
                if (input.equals("t")
                        || input.equals("true")
                        || input.equals("y")
                        || input.equals("yes")
                        || input.equals("ok")
                        || input.equals("on")
                        || input.equals("1")) {
                    result = true;
                }
            }
        } catch (Exception e) {
            Log.d(TAG, e.getMessage());
        }

        return result;
    }

    public static boolean isFalse(String input) {
        return !isTrue(input);
    }
}

File: app/src/main/java/com/rr/tvstarterkit/data/preference/BooleanUtils.kt
package com.rr.tvstarterkit.data.preference

import android.text.TextUtils
import android.util.Log
import java.lang.Exception
import java.util.*

object BooleanUtils {
    private const val TAG = "BooleanUtils"

    /**
     * @param input
     * @return true if input is one of (T, True, TRUE, Y, yes, YES, 1, OK, ok, on, ON)
     */
    @JvmStatic
    fun isTrue(input: String): Boolean {
        var input = input
        var result = false
        try {
            if (!TextUtils.isEmpty(input)) {
                input = input.lowercase(Locale.getDefault())
                if (input == "t" || input == "true" || input == "y" || input == "yes" || input == "ok" || input == "on" || input == "1") {
                    result = true
                }
            }
        } catch (e: Exception) {
            Log.d(TAG, e.message!!)
        }
        return result
    }

    fun isFalse(input: String): Boolean {
        return !isTrue(input)
    }
}